asv_bench/benchmarks/utils/data_shapes.py

```diff
@@ -0,0 +1,186 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Define data shapes."""
+
+import os
+import json
+
+from .compat_with_old_modin import ASV_USE_BACKEND, ASV_DATASET_SIZE
+
+RAND_LOW = 0
+RAND_HIGH = 1_000_000_000 if ASV_USE_BACKEND == "omnisci" else 100
+
+BINARY_OP_DATA_SIZE = {
+    "big": [
+        ((5000, 5000), (5000, 5000)),
+        # the case extremely inefficient
+        # ((20, 500_000), (10, 1_000_000)),
+        ((500_000, 20), (1_000_000, 10)),
+    ],
+    "small": [
+        ((250, 250), (250, 250)),
+        ((20, 10_000), (10, 25_000)),
+        ((10_000, 20), (25_000, 10)),
+    ],
+}
+UNARY_OP_DATA_SIZE = {
+    "big": [
+        (5000, 5000),
+        # the case extremely inefficient
+        # (10, 1_000_000),
+        (1_000_000, 10),
+    ],
+    "small": [
+        (250, 250),
+        (10, 10_000),
+        (10_000, 10),
+    ],
+}
+
+
+OMNISCI_BINARY_OP_DATA_SIZE = {
+    "big": [
+        ((500_000, 20), (1_000_000, 10)),
+    ],
+    "small": [
+        ((10_000, 20), (25_000, 10)),
+    ],
+}
+OMNISCI_UNARY_OP_DATA_SIZE = {
+    "big": [
+        (1_000_000, 10),
+    ],
+    "small": [
+        (10_000, 10),
+    ],
+}
+OMNISCI_SERIES_DATA_SIZE = {
+    "big": [
+        (10_000_000, 1),
+    ],
+    "small": [
+        (100_000, 1),
+    ],
+}
+
+BINARY_SHAPES = (
+    OMNISCI_BINARY_OP_DATA_SIZE[ASV_DATASET_SIZE]
+    if ASV_USE_BACKEND == "omnisci"
+    else BINARY_OP_DATA_SIZE[ASV_DATASET_SIZE]
+)
+UNARY_SHAPES = (
+    OMNISCI_UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE]
+    if ASV_USE_BACKEND == "omnisci"
+    else UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE]
+)
+OMNISCI_SERIES_SHAPES = OMNISCI_SERIES_DATA_SIZE[ASV_DATASET_SIZE]
+
+DEFAULT_GROUPBY_NGROUPS = {
+    "big": [100, "huge_amount_groups"],
+    "small": [5],
+}
+GROUPBY_NGROUPS = DEFAULT_GROUPBY_NGROUPS[ASV_DATASET_SIZE]
+
+DEFAULT_CONFIG = {
```

Comment:
I propose to reverse the logic for defining the list, as in current form it has a looooot of lines which look very similar.
I propose to transpose it like (untested)
```python
_DEFAULT_CONFIG_T = [
    (UNARY_SHAPES, ["TimeGroupByMultiColumn", "TimeGroupByDefaultAggregations", "TimeJoin", ......]),
    # more stuff like that
]
DEFAULT_CONFIG = {}
for _shape, _names in _DEFAULT_CONFIG_T:
    DEFAULT_CONFIG.update({_name: _shape for _name in _names})
```